midterm/app/Main.hs

```diff
@@ -4,8 +4,17 @@ import Directory
 
 -- Implement a function that asks the user for the directory name and the sorting type, constructs the tree and then displays it accordingly.
 run :: IO () 
-run = undefined 
+run = do
+  putStrLn "Enter the directory path:"
+  dirPath <- getLine
+  putStrLn "Enter sorting type (default/alphabetical):"
+  sortingType <- getLine
+  tree <- buildTree dirPath
+  let displayFunc = case sortingType of
+          "alphabetical" -> alphabetisedDisplayTree
+          _              -> defaultDisplayTree
```

Comment:
Using a wildcard here is error-prone: it's easy to make a typo in the word "alphabetical" and end up with an incorrect option. Better to match on "alphabetical", "default" and then in a catch-all case notify the user about the two correct options.